pkg/server/http/ipfs.go

```diff
@@ -241,17 +241,20 @@ func ipfsHandler(lassie *lassie.Lassie, cfg HttpServerConfig) func(http.Response
 				return
 			}
 
+			header.Lock()
 			if header.Metrics != nil {
 				for _, m := range header.Metrics {
 					if m.Name == string(re.Phase()) {
 						if m.Extra == nil {
 							m.Extra = map[string]string{}
 						}
 						m.Extra[string(re.Code())] = fmt.Sprintf("%d", re.Time().Sub(re.PhaseStartTime()))
+						header.Unlock()
 						return
 					}
 				}
 			}
+			header.Unlock()
 
 			metric := header.NewMetric(string(re.Phase()))
 			metric.Duration = re.Time().Sub(re.PhaseStartTime())
```

Comment:
Do you think we need the lock for this metrics write as well?

Reply:
no, I don't think so since we only touch `Duration` once - i.e. when `if header.Metrics != nil` and `if m.Name == string(re.Phase())` then we get the early `return` above. When those don't hold true, we make a new one and set `Duration` and this should be done only once per Phase `Metric` so should be safe (a tiny bit 🤞 but I don't see how this wouldn't hold true).